oak/server/rust/oak_runtime/src/runtime/tests.rs

```diff
@@ -43,11 +43,18 @@ fn run_node_body(node_label: &Label, node_body: Box<NodeBody>) {
 
     struct TestNode {
         node_body: Box<NodeBody>,
+        done: Arc<(Mutex<bool>, Condvar)>,
     };
 
     impl crate::node::Node for TestNode {
         fn run(self: Box<Self>, runtime: RuntimeProxy, _handle: oak_abi::Handle) {
             let _ = (self.node_body)(runtime);
+
+            // Trigger the condition variable when the test is done.
+            let (lock, cvar) = &*self.done;
+            let mut started = lock.lock().unwrap();
+            *started = true;
```

Comment:
This works a bit magically because in these tests, the node body has access to the test context and can cross that boundary to notify the synchronization mechanism; but say that the node body was fixed (e.g. the logger node, or some wasm module), or we ran it under more realistic circumstances, then we would not be able to cross that boundary. I was thinking it would be more maintainable instead to have `node_start_instance` return the join handle of the node, or the NodeId (though this may be problematic if we treat the NodeId as a "secret"), then the test would wait on that. Or just make the test wait for the termination of the runtime itself (which I think we should do anyways). WDYT?

Reply:
Good spot, that's a much simpler change that achieves the same effect.  (BTW, the tests do now wait for the runtime to stop, at line 95 below.)